include/vcpkg/base/json.h

```diff
@@ -120,7 +120,13 @@ namespace vcpkg::Json
         static Value boolean(bool) noexcept;
         static Value integer(int64_t i) noexcept;
         static Value number(double d) noexcept;
-        static Value string(std::string s) noexcept;
+        template<class StringLike>
+        static Value string(StringLike&& s) noexcept
+        {
+            return string(StringView(s).to_string());
+        }
+        static Value string(std::string&& s) noexcept;
```

Comment:
I think we should reorder this one before the template (and also on 212). Yes I know C++ makes it not matter but I think it's clearer to read if the overload we intend to call is right before.